File: main/src/main/java/lavaplayer/source/youtube/format/OfflineYoutubeTrackFormatExtractor.java
package lavaplayer.source.youtube.format;

import lavaplayer.source.youtube.YoutubeSignatureResolver;
import lavaplayer.source.youtube.YoutubeTrackFormat;
import lavaplayer.source.youtube.YoutubeTrackJsonData;
import lavaplayer.tools.io.HttpInterface;

import java.util.List;

public interface OfflineYoutubeTrackFormatExtractor extends YoutubeTrackFormatExtractor {
    List<YoutubeTrackFormat> extract(YoutubeTrackJsonData data);

    @Override
    default List<YoutubeTrackFormat> extract(
        YoutubeTrackJsonData data,
        HttpInterface httpInterface,
        YoutubeSignatureResolver signatureResolver
    ) {
        return extract(data);
    }
}

File: main/src/main/java/lavaplayer/source/youtube/format/OfflineYoutubeTrackFormatExtractor.kt
package lavaplayer.source.youtube.format

import lavaplayer.source.youtube.YoutubeSignatureResolver
import lavaplayer.source.youtube.YoutubeTrackFormat
import lavaplayer.source.youtube.YoutubeTrackJsonData
import lavaplayer.tools.io.HttpInterface

abstract class OfflineYoutubeTrackFormatExtractor : YoutubeTrackFormatExtractor {
    abstract fun extract(data: YoutubeTrackJsonData): List<YoutubeTrackFormat>

    override fun extract(
        response: YoutubeTrackJsonData,
        httpInterface: HttpInterface,
        signatureResolver: YoutubeSignatureResolver
    ): List<YoutubeTrackFormat> =
        extract(response)
}
